nixos/modules/hardware/gkraken.nix

```diff
@@ -0,0 +1,18 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.hardware.gkraken;
+in
+{
+  options.hardware.gkraken = {
+    enable = mkEnableOption "gkraken's udev rules for NZXT AIO liquid coolers";
+  };
+
+  config = mkIf cfg.enable {
+    services.udev.packages = with pkgs; [
+      gkraken
```

Comment:
Maybe move them to an extra output.

Reply:
I tried doing that & adding `gkraken.udev` instead but it refused to install the udev rules like that.

Reply:
WDYM it refused to install the rules? Did you manually copy them to the output? I don't think it is done automatically for udev.

Reply:
I added `outputs = [ "out" "udev" ];` and replaced the `$out`s in the postInstall script to `$udev`. The file was written to the `udev` output properly. When I went and tried to include `gkraken.udev` in my system's `services.udev.packages` however it would fail to find the udev rule.

```
Generating hwdb database...
Adding rules for package /nix/store/03vdwkhxalw48r9xi4x73092b4mmicdl-gkraken-1.1.6
Adding rules for package /nix/store/rzh2yms7jnjh84gydk7ncilp1bcfkjnn-elementary-settings-daemon-3.34.1
Copying /nix/store/rzh2yms7jnjh84gydk7ncilp1bcfkjnn-elementary-settings-daemon-3.34.1/lib/udev/rules.d/61-gnome-settings-daemon-rfkill.rules to /nix/store/jl4l1q00bivng4yrnmv8cazjg33x183b-udev-rules/61-gnome-settings-daemon-rfkill.rules
```

And trying to launch graken would give me an error about missing permissions to my AIO. Changing everything back to `$out` and `gkraken` made it work again.

```
Generating hwdb database...
building '/nix/store/fxibdkahcns656i5ys5jak4yvkkc8549-udev-rules.drv'...
Adding rules for package /nix/store/wkblvw1jiicxv8ilz5c3vxzwspm4dps9-gkraken-1.1.6
Copying /nix/store/wkblvw1jiicxv8ilz5c3vxzwspm4dps9-gkraken-1.1.6/lib/udev/rules.d/60-gkraken.rules to /nix/store/x3gmajd6hbgpvignqrpvhipacpk5f38p-udev-rules/60-gkraken.rules
Adding rules for package /nix/store/rzh2yms7jnjh84gydk7ncilp1bcfkjnn-elementary-settings-daemon-3.34.1
Copying /nix/store/rzh2yms7jnjh84gydk7ncilp1bcfkjnn-elementary-settings-daemon-3.34.1/lib/udev/rules.d/61-gnome-settings-daemon-rfkill.rules to /nix/store/x3gmajd6hbgpvignqrpvhipacpk5f38p-udev-rules/61-gnome-settings-daemon-rfkill.rules
```

Reply:
Hmmm.... Not sure why this does not work. Sounds correct what you are doing.